mypy/semanal_typeargs.py

```diff
@@ -102,6 +102,8 @@ def visit_type_alias_type(self, t: TypeAliasType) -> None:
             # If there was already an error for the alias itself, there is no point in checking
             # the expansion, most likely it will result in the same kind of error.
             get_proper_type(t).accept(self)
+            if t.alias is not None:
+                t.alias.accept(self)
```

Comment:
This looks like a duplicate work, did you try deleting the accept just above? (if you delete it, don't forget to update the comment)

Reply:
OK, this is kind of unfortunate. We already accept args in isolation at the very start (see super call), then we accept expansion, and finally the target in isolation. But I guess this is the price to pay for having both:
* Support for unrestricted type vars in alias definitions
* Doing some in-place patching in this visitor